### ISSUE TITLE
Extend actor computation expression

### ISSUE THREAD

**Issue by Horusiath** [Horusiath/Akkling#10]
Right now our `actor { ...}` computation expression follows free monad pattern. This could be extended.

Example:

```fsharp
let rec loop () =
    actor {
        let! msg = mailbox.Receive ()
        match msg with
        | "persist" -> persist! (loop()) [msg] // like return! loop ()... but with Persistence post-effect
        | "stop" -> stop!                      // like return! 0
        | _ -> unhandled! msg                  // like return! loop ()... but with Unhandled message post-effect
    }
```

where `stop!`, `persist!` and `unhandled!` would be custom actor builder methods, just like now we have `return!` and `let!`.

**Comment by Horusiath** [comment]
Finally, this has been solved differently using effects - from now on Behavior allways returns continuation (the next recursive behavior call) or effect. Effect itself is an interface, that provides some operation after it's returned from behavior.

Rigth now Akkling provides following effects:

- `Stop` results in actor to stopping.itself.
- `Unhandled` causes current message to be unhandled and returns to behavior loop.
- `Persist` (only in Akkling.Persistence) causes sequence of objects to be stored in the persistent journal and return back to behavior loop.
- `Defer` (only in Akkling.Persistence) invokes behavior loop after events have been saved.

**Comment by Horusiath** [comment]
Closed with release 0.2
